core/src/main/java/hudson/Functions.java

```diff
@@ -758,6 +758,19 @@ public static boolean isMultiline(String s) {
         return s.indexOf('\r') >= 0 || s.indexOf('\n') >= 0;
     }
 
+    /**
+     * Partially encodes a string into UTF-8 encoding. It uses the {@link hudson.Util} encode method to escape non-ASCII characters in URL.
+     * <pre>
+     * Input example  1: !"£$%^&amp;*()_+}{:@~?&gt;&lt;|¬`,./;'#[]-=
+     * Output example 1: !"%C2%A3$%^&amp;*()_+}{:@~?&gt;&lt;|%C2%AC`,./;'#[]-=%20
```

Comment:
I'd put the space somewhere in the middle. As a trailing space it'll just keep getting erased.
```suggestion
     * Input example  1: !"£$%^&amp;*()_+}{:@~?&gt;&lt;|¬`,./;'#[]- =
     * Output example 1: !"%C2%A3$%^&amp;*()_+}{:@~?&gt;&lt;|%C2%AC`,./;'#[]-%20=
```